apps/coordinator/ts/health/health.service.ts

```
@@ -0,0 +1,164 @@
+import { EMode } from "@maci-protocol/sdk";
+import { Injectable } from "@nestjs/common";
+import { formatEther } from "ethers";
+
+import { accessSync, constants, statSync } from "fs";
+import path from "path";
+
+import { ESupportedNetworks, getSigner } from "../common";
+import { FileService } from "../file/file.service";
+
+import { ICheckRapidsnark, ICheckWalletFunds, ICheckZkeysDirectory, IHealthCheckResponse } from "./types";
+
+/**
+ * Health service functions to check the status of the coordinator service.
+ */
+@Injectable()
+export class HealthService {
+  /**
+   * Create a new instance of the HealthService.
+   *
+   * @param fileService - file service
+   */
+  constructor(private readonly fileService: FileService) {}
+
+  /**
+   * Check if rapidsnark is available and executable.
+   * @returns the rapidsnark executable path and its executability status
+   */
+  checkRapidsnark(): ICheckRapidsnark {
+    const rapidsnarkExecutablePath = process.env.COORDINATOR_RAPIDSNARK_EXE || "";
+    let rapidsnarkIsExecutable = false;
+
+    try {
+      // Check if it's a file and has execute permissions
+      // eslint-disable-next-line no-bitwise
+      accessSync(rapidsnarkExecutablePath, constants.F_OK | constants.X_OK);
+      const stats = statSync(rapidsnarkExecutablePath);
+
+      rapidsnarkIsExecutable = stats.isFile();
+    } catch (error) {
+      rapidsnarkIsExecutable = false;
+    }
```

Why the error is silenced (applied to all catches)?